kube/src/api/subresource.rs

```diff
@@ -279,50 +279,61 @@ impl Default for AttachParams {
 
 #[cfg(feature = "ws")]
 impl AttachParams {
+    /// Default parameters for an tty exec with stdin and stdout
+    pub fn interactive_tty() -> Self {
+        Self {
+            stdin: true,
+            stdout: true,
+            stderr: false,
+            tty: true,
+            ..Default::default()
+        }
+    }
+
     /// Specify the container to execute in.
-    pub fn container<T: Into<String>>(&mut self, container: T) -> &mut Self {
+    pub fn container<T: Into<String>>(mut self, container: T) -> Self {
         self.container = Some(container.into());
         self
     }
 
     /// Set `stdin` field.
-    pub fn stdin(&mut self, enable: bool) -> &mut Self {
+    pub fn stdin(mut self, enable: bool) -> Self {
```

Comment:
Is there a reason to consume it?

This change will make it awkward for something like this:
```rust
let mut ap = AttachParams::default();
if some_option {
    ap.stdin(true); // moves `ap`. need `ap = ap.stdin(true);`
}
let mut attached = pods.exec("example", command, &ap).await?; // error borrow of moved
```

Reply:
If you don't consume it, it becomes more awkward to do in-place builders if you want to store them:

```rust
        let ap = AttachParams::default()
                    .stdin(true)
                    .stdout(true)
                    .stderr(false)
                    .tty(true);
```
```
57 |         let ap = AttachParams::default()
   |                  ^^^^^^^^^^^^^^^^^^^^^^^ creates a temporary which is freed while still in use
...
61 |                     .tty(true);
   |                               - temporary value is freed at the end of this statement
...
66 |                 &ap,
   |                 --- borrow later used here
```

but it's a tradeoff. the `&mut self` is nicer when you have tons of options. Though you could also be explicit about the booleans and do:

```rust
let stdin_enabled = some_option;
let ap = AttachParams::default().stdin(stdin_enabled);
````

and since that workaround exists, moving builder I feel is a little cleaner, at least for __smaller__ builders.

Reply:
Ah, good point. I agree with you :+1: